.github/workflows/prepare-transaction.yml

```
@@ -0,0 +1,207 @@
+
+          if [ "$chainlist_status_code" == "404" ]; then
+            echo "Chain $CHAIN_ID is not in chainlist."
+            echo "COMMENT_OUTPUT="$ERROR_MSG_CHAINLIST_FAILURE"" >> $GITHUB_ENV
+            exit 1
+          fi
+
+      - name: Estimate Transaction and Check Pre-fund
+        if: success() && ${{ steps.validate_description.outputs.rpc_url != null }}
+        env:
+          RPC_URL: ${{ steps.validate_description.outputs.rpc_url }}
+        run: |
+          json_request='[
+              {
+                  "jsonrpc": "2.0",
+                  "method": "eth_gasPrice",
+                  "params": [],
+                  "id": 1
+              },
+              {
+                  "jsonrpc": "2.0",
+                  "method": "eth_estimateGas",
+                  "params": [
+                      {
+                          "from": "'$FACTORY_DEPLOYER_ADDRESS'",
+                          "data": "'$FACTORY_BYTECODE'"
+                      }
+                  ],
+                  "id": 2
+              }
+          ]'
+
+          response=$(curl "$RPC_URL" --location --header 'Content-Type: application/json' --data "$json_request")
+
+          if jq -e . >/dev/null 2>&1 <<< "$response"; then
+            gas_price=$(jq -r '.[0].result' <<< "$response")
+            echo "Current gas price: $gas_price"
+            gas_limit=$(jq -r '.[1].result' <<< "$response")
+            echo "Estimated deployment gas limit: $gas_limit"
+
+            if [[ "$gas_price" == "null" || "$gas_limit" == "null" ]]; then
+              echo "COMMENT_OUTPUT="$ERROR_MSG_PREFUND_CHECK"" >> $GITHUB_ENV
+              exit 1
+            fi
+
+            # We multiply the gas limit by 1.4, just like the deployment script does it
+            gas_limit=$(( $gas_limit * 14 / 10 ))
+            expected_prefund=$(($gas_limit * $gas_price))
+
+            echo "Expected pre-fund: $expected_prefund"
+
+            deployer_address_balance=$(curl $RPC_URL --location --header \
+            'Content-Type: application/json' --data '{
+                          "jsonrpc": "2.0",
+                          "method": "eth_getBalance",
+                          "params": ["'$FACTORY_DEPLOYER_ADDRESS'", "latest"],
+                          "id": 5413
+                      }')
+
+            if jq -e . >/dev/null 2>&1 <<< "$deployer_address_balance"; then
+              deployer_address_balance=$(jq -r '.result' <<< "$deployer_address_balance")
+              deployer_address_balance=${deployer_address_balance#0x}
+              deployer_address_balance=$((16#$deployer_address_balance))
+              echo "Deployer address balance: $deployer_address_balance"
+
+              if [ "$deployer_address_balance" -lt "$expected_prefund" ]; then
+                echo "COMMENT_OUTPUT=**⛔️ Error:**<br/>The deployer address is not pre-funded. Please send $expected_prefund wei to $FACTORY_DEPLOYER_ADDRESS and try again." >> $GITHUB_ENV
```

Going back to this - should this be quoted?
```suggestion
                echo "COMMENT_OUTPUT=\"**⛔️ Error:**<br/>The deployer address is not pre-funded. Please send $expected_prefund wei to $FACTORY_DEPLOYER_ADDRESS and try again.\"" >> $GITHUB_ENV
```

If not, you can probably ignore my other suggestions from before and turn

```bash
echo "COMMENT_OUTPUT="$ERROR_MSG_PREFUND_CHECK"" >> $GITHUB_ENV 
```


Into
```bash
echo "COMMENT_OUTPUT=$ERROR_MSG_PREFUND_CHECK" >> $GITHUB_ENV 
```

Or is there some weird variable expansion magic that is happening which needs `ERROR_MSG_PREFUND_CHECK` to be expanded unquoted?